great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -40,8 +37,7 @@ class DataAssistant(ABC):
 
     data_assistant: DataAssistant = VolumeDataAssistant(
         name="my_volume_data_assistant",
-        batch_request=batch_request,
-        data_context=context,
+        validator=validator,
```

Comment:
Will we eventually get to a place where `validator` is not needed here, or is instantiating a `validator` now a required step prior to instantiating a `DataAssistant`?

Reply:
@NathanFarmer The (now private) `Validator` is needed to build the profiler, supply the loaded `Batch` objects, etc.  Of course the specifics are subject to change once the `BatchManager` and/or `BatchMetricStore` come online.  Thanks.

Reply:
We will still be able to do this though right?

```
validator = context.get_validator(
    batch_request=BatchRequest(**batch_request),
    expectation_suite_name=expectation_suite_name
)
```

Reply:
@NathanFarmer I am not sure.  It might go in the direction of some other entity grabbing the `Batch` objects of its own choosing, then (optionally) creating an `ExpectationSuite` using a utility it deems appropriate, and then instantiating the `Validator`.  But what `Validator` is today and what it will be in the future may not be the same.  For example, there have been conversations about unbundling the `Validator` into these parts:
* MetricsCalculator
* ExpectationExecutor
* Possibly also containing ExpectationSuiteManager (or leaving that to a different component)

But today, `Validator` does all of the above; in addition, it also, rather unfortunately, maintains the copy of the cache of `Batch` objects, which it must keep in sync with the same cache in `ExecutionEngine` base class.

Other ideas are welcome.  Thanks! :-)